_posts/2021-11-05-r-6.0.0.md

```diff
@@ -0,0 +1,207 @@
+
+A big improvement in 6.0.0 is that `arrow` can now install in a fully offline mode. The R package now includes the C++ source, so it does not need to be downloaded at build time. This does not include optional dependencies like compression libraries, the AWS SDK for accessing data in S3, and more. For folks who need to install Arrow on an airgapped server with all of those features, we have included a helper function to download and assemble a "fat" pacakge that contains everything that would be downloaded lazily at build time.
+The function `create_package_with_all_dependencies()` can be run from a computer that does have access to the internet, and creates a fat-source package which can then be transferred and installed on a server without connectivity. This helper is also available on GitHub without installing the arrow package. For more installation [see the docs](https://arrow.apache.org/docs/r/articles/install.html#offline-installation).
+
+Another installation change is that we've changed the source build to fail cleanly if the C++ library is not found or cannot be built. Previously, if the C++ library failed to build, you would get a successful R package installation, but the package wouldn't do anything useful, it would just tell you to reinstall. This was helpful back in the early days of the package when we weren't confident it would build everywhere that CRAN checked, but we now have much more experience (and extensive testing). In recent months this failure mode caused more confusion than it was worth, and it led many people to think that after you install arrow, you always have to `install_arrow()` again.
+
+# Thanks
+
+This is a significant milestone for Arrow, and the R package specifically, and there is much gratitude to go around. In the 6.0.0 release, there were 77 individuals who contributed to Arrow, many of whom did the heavy lifting in the C++ library to make the new dataset query features a reality. Specifically in the R, we wanted to acknowledge Phillip Cloud, Dewey Dunnington, Dragoș Moldovan-Grünfeld, Matt Peterson, and Percy Camilo Triveño Aucahuasi for their
+their first contributions the to the R package. And a special thanks goes to Karl Dunkle Werner for the hard work on the offline package build!
+
+We also want to thank you in advance for your help. For this release of the Arrow query engine, we've focused our effort on getting the core functionality implemented. (In fact, this first release is something of an R-exclusive: bindings for these features haven't yet been added to pyarrow, the Python Arrow library!) By focusing on the essentials, it means that there are a number of performance optimizations plan to do but didn't have time for in this release--and there are surely more issues to improve that we don't yet know. We are eager for your feedback: please [let us know](https://issues.apache.org/jira/browse/ARROW) of any issues you encounter so that we can improve these for our next release.
```

Comment:
```suggestion
We also want to thank you in advance for your help. For this release of the Arrow query engine, we've focused our effort on getting the core functionality implemented. (In fact, this first release is something of an R-exclusive: bindings for these features haven't yet been added to pyarrow, the Python Arrow library!) By focusing on the essentials, it means that there are a number of performance optimizations we plan to do but didn't have time for in this release--and there are surely more issues to improve that we don't yet know. We are eager for your feedback: please [let us know](https://issues.apache.org/jira/browse/ARROW) of any issues you encounter so that we can improve these for our next release.
```